pkg/recommender/metrics/provider.go

```diff
@@ -2,15 +2,17 @@ package metrics
 
 import (
 	"context"
-	"github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/scylladb/go-log"
+	"time"
 )
 
 type Provider interface {
-	FetchMetric(ctx context.Context, metric string, labels map[string]string) (float64, error)
+	Query(ctx context.Context, expression string) (bool, error)
```

Comment:
It's a good practice to write short description of exposed functions in comment. Especially for exposed interfaces.